securedrop/loaddata.py

```diff
@@ -0,0 +1,419 @@
+    db.session.add(author_seen_reply)
+
+    if journalist_who_saw:
+        other_seen_reply = SeenReply(
+            reply_id=reply.id, journalist_id=journalist_who_saw.id
+        )
+        db.session.add(other_seen_reply)
+
+    db.session.commit()
+
+
+def add_source() -> Tuple[Source, str]:
+    """
+    Adds a single source.
+    """
+    codename = current_app.crypto_util.genrandomid()
+    filesystem_id = current_app.crypto_util.hash_codename(codename)
+    journalist_designation = current_app.crypto_util.display_id()
+    source = Source(filesystem_id, journalist_designation)
+    source.pending = False
+    db.session.add(source)
+    db.session.commit()
+
+    # Create source directory in store
+    os.mkdir(current_app.storage.path(source.filesystem_id))
+
+    # Generate source key
+    current_app.crypto_util.genkeypair(source.filesystem_id, codename)
+
+    return source, codename
+
+
+def star_source(source: Source) -> None:
+    """
+    Adds a SourceStar record for the source.
+    """
+    star = SourceStar(source, True)
+    db.session.add(star)
+    db.session.commit()
+
+
+def load(args: argparse.Namespace) -> None:
+    """
+    Populate the database.
+    """
+    if not os.environ.get("SECUREDROP_ENV"):
+        os.environ["SECUREDROP_ENV"] = "dev"
+
+    app = journalist_app.create_app(config)
+    with app.app_context():
+        try:
+            default_journalist = add_journalist("journalist", is_admin=True)
+        except IntegrityError as e:
+            db.session.rollback()
+            if "UNIQUE constraint failed: journalists." in str(e):
+                default_journalist = Journalist.query.filter_by(username="journalist").one()
+            else:
+                raise e
+
+        try:
+            dellsberg = add_journalist("dellsberg")
+        except IntegrityError as e:
+            db.session.rollback()
+            if "UNIQUE constraint failed: journalists." in str(e):
+                dellsberg = Journalist.query.filter_by(username="dellsberg").one()
+            else:
+                raise e
+
+        try:
+            journalist_to_be_deleted = add_journalist(
+                username="clarkkent", first_name="Clark", last_name="Kent"
+            )
+        except IntegrityError as e:
+            db.session.rollback()
+            if 'UNIQUE constraint failed: journalists.' in str(e):
+                journalist_to_be_deleted = Journalist.query.filter_by(username="clarkkent").one()
+            else:
+                raise e
+
+        for i in range(1, args.journalist_count + 1):
+            add_journalist()
+
+        starred_sources_count = int(args.source_count * args.source_star_fraction)
+        replied_sources_count = int(args.source_count * args.source_reply_fraction)
+
+        for i in range(1, args.source_count + 1):
+            source, codename = add_source()
+
+            for _ in range(args.messages_per_source):
+                journalist_who_saw = random.choice(
```

Comment:
@creviera - this is why seen/unseen varies

Reply:
yup, random.choice is not going to be predictable

Reply:
i think this pr should try to keep the current behavior with seen/unseen made in the PR: https://github.com/freedomofpress/securedrop/pull/5580

i use this script everyday so it's nice to have predictability with creating seen data, although perhaps we should have more variation with file attachments and last_updated, but this can be done outside of this pr.